dali/operators/segmentation/select_masks.cc

```diff
@@ -0,0 +1,236 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/segmentation/select_masks.h"
+
+#include <utility>
+#include "dali/kernels/common/type_erasure.h"
+#include "dali/core/static_switch.h"
+
+namespace dali {
+
+DALI_SCHEMA(segmentation__SelectMasks)
+    .DocStr(R"(Selects a subset of polygons by their mask ids.
+
+The operator expects three inputs describing multiple segmentation mask polygons belonging to different mask ids and 
+a list of selected mask ids.
+
+Each sample can contain several polygons belonging to different masks, and each polygon can be composed by an arbitrary 
+number of vertices (at least 3). The masks polygons are described  by the inputs ``polygons`` and ``vertices`` and 
+the operator produces output ``polygons`` and ``vertices`` where only the polygons associated with the selected 
+masks are present.
+
+.. note::
+  
+  The format of ``polygons`` and ``vertices`` is the same as ``mask_meta`` and ``mask_coords`` outputs from COCOReader.
+
+**Examples:**
+
+Let us assume the following input mask, where symbolic coordinates are used for a clearer example::
+
+    polygons = [[0, 0, 3], [1, 3, 7], [2, 7, 10]]
+    vertices = [[x0, y0], [x1, y1], [x2, y2], [x3, y3], [x4, y4], [x5, y5], [x6, y6], [x7, y7], [x8, y8], [x9, y9]]
+
+Example 1: Selecting a single mask with id ``1``, maintaining the original id::
+
+    mask_ids = [1], ``reindex_masks`` = False
+    out_polygons = [[1, 0, 4]]
+    out_vertices = [[x3, y3], [x4, y4], [x5, y5], [x6, y6]]
+
+Example 2: Selecting two out of the three masks, replacing the mask ids with the indices at which
+they appeared in ``mask_ids`` input::
+
+    mask_ids = [2, 0]
+    reindex_masks = True
+    out_polygons = [[0, 3, 6], [1, 0, 3]]
+    out_vertices = [[x0, y0], [x1, y1], [x2, y2], [x7, y7], [x8, y8], [x9, y9]]
+)")
+    .NumInput(3)
+    .NumOutput(2)
+    .InputDox(0, "mask_ids", "1D TensorList of int",
+              R"code(List of identifiers of the masks to be selected.)code")
+    .InputDox(1, "polygons", "2D TensorList of int",
```

Comment:
I started wondering if we should extend this to 3D as well.

Reply:
3D is supported. Here 2D refers to the shape of the data:

[[x0, y0, (z0, ...)], ..., [xN, yN, (zN, ...)]